lib/pennyworth/cli.rb

```diff
@@ -0,0 +1,339 @@
+            result << "Error output:\n"
+            result << "#{e.stderr}\n"
+          end
+
+          if e.stdout && !e.stdout.empty?
+            result << "Standard output:\n"
+            result << "#{e.stdout}\n\n"
+          end
+
+          if e.backtrace && !e.backtrace.empty?
+            result << "Backtrace:\n"
+            result << "#{e.backtrace.join("\n")}\n\n"
+          end
+          STDERR.puts result
+          exit 1
+        else
+          raise
+        end
+      end
+      true
+    end
+
+    def self.settings
+      @@settings
+    end
+
+    def self.settings= s
+      @@settings = s
+    end
+
+    desc "Prepare system for running Pennyworth"
+    long_desc <<-LONGDESC
+      Prepare the system to be able to run virtual machines via Pennyworth. This
+      encapsulates package installation as well as set up of required tools such
+      as Vagrant and libvirt.
+
+      See https://github.com/SUSE/pennyworth#installation for more information.
+    LONGDESC
+    command :setup do |c|
+      c.action do |global_options,options,args|
+        SetupCommand.new.execute
+      end
+    end
+
+    desc "Show status of virtual machine"
+    long_desc <<-LONGDESC
+      Show status of specified virtual machine. Use `list` to get a list of all
+      available machines.
+    LONGDESC
+    arg_name "VM_NAME"
+    command :status do |c|
+      c.action do |global_options,options,args|
+        vm_name = args.shift
+        VagrantCommand.setup_environment(@@settings.vagrant_dir)
+        StatusCommand.new.execute(vm_name)
+      end
+    end
+
+    desc "Build base images"
+    long_desc <<-LONGDESC
+      Build base images used by Vagrant. If no specific name is given all base
+      images are built. The images have to be imported with `import-base` to be
+      available to be run.
+    LONGDESC
+    arg_name "IMAGE_NAME", :optional
+    command "build-base" do |c|
+      c.flag [:kiwi_tmp_dir, :k], :type => String, :required => false,
+        :desc => "Temporary KIWI directory for building the Vagrant box.",
+        :arg_name => "KIWI-TMP-Dir"
+      c.action do |global_options,options,args|
+        image_name = args.shift
+        tmp_dir = options[:kiwi_tmp_dir] || "/tmp/pennyworth-kiwi-builds"
+        BuildBaseCommand.new(Cli.settings.kiwi_dir).execute(tmp_dir, image_name)
+      end
+    end
+
+    desc "Import base images"
+    long_desc <<-LONGDESC
+      Import base images used by Vagrant. If no specific name is given all base
+      images are imported. Base images have to be built with `build-base` or
+      imported from a remote location specified with the `--url` option.
+    LONGDESC
+    arg_name "IMAGE_NAME", :optional
+    command "import-base" do |c|
+      c.flag [:url, :u], :type => String, :required => false,
+        :desc => "URL of the remote server where the images will be imported from", :arg_name => "URL"
+      c.action do |global_options,options,args|
```

Comment:
Unused block argument - `global_options`. If it's necessary, use `_` or `_global_options` as an argument name to indicate that it won't be used.<br>Space missing after comma.